lib/api/dsl/transform/canonical.js

```diff
@@ -32,60 +33,192 @@ function Canonical () {
    * @return {Array} resolving to a simplified filters array
    * @throws if espresso is unable to normalize the provided filters
    */
-  this.convert = function convert (filters) {
-    var
-      conditions,
-      pla,
-      normalized,
-      result = [],
-      clone = _.cloneDeep(filters);
+  convert (filters) {
+    let
+      result = [];
 
     if (Object.keys(filters).length === 0) {
       return [[{'everything': true}]];
     }
 
-    conditions = extractConditions(clone);
-    pla = buildPLA(clone, conditions.length);
-    normalized = espresso(pla);
+    const conditions = extractConditions(this._cloneFilters(filters));
```

Comment:
This would make the function quite hairy. 

The clone is just there to prevent messing up with the internal reformatting that is done later on by the actual canonization.